crates/js-sys/src/lib.rs

```diff
@@ -2848,19 +2848,163 @@ extern "C" {
+    #[wasm_bindgen]
+    extern "C" {
+        /// The `WebAssembly.RuntimeError()` constructor creates a new WebAssembly
+        /// `RuntimeError` object — the type that is thrown whenever WebAssembly
+        /// specifies a trap.
+        ///
+        /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/RuntimeError
+        #[wasm_bindgen(extends = Error, js_namespace = WebAssembly)]
+        #[derive(Clone, Debug)]
+        pub type RuntimeError;
+
+        /// The `WebAssembly.RuntimeError()` constructor creates a new WebAssembly
+        /// `RuntimeError` object — the type that is thrown whenever WebAssembly
+        /// specifies a trap.
+        ///
+        /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/RuntimeError
+        #[wasm_bindgen(constructor, js_namespace = WebAssembly)]
+        pub fn new(message: &str) -> RuntimeError;
+    }
+
+    // WebAssembly.Module
+    #[wasm_bindgen]
+    extern "C" {
+        /// A `WebAssembly.Module` object contains stateless WebAssembly code
+        /// that has already been compiled by the browser and can be
+        /// efficiently shared with Workers, and instantiated multiple times.
+        ///
+        /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Module
+        #[wasm_bindgen(js_namespace = WebAssembly, extends = Object)]
+        #[derive(Clone, Debug)]
+        pub type Module;
+
+        /// A `WebAssembly.Module` object contains stateless WebAssembly code
+        /// that has already been compiled by the browser and can be
+        /// efficiently shared with Workers, and instantiated multiple times.
+        ///
+        /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Module
+        #[wasm_bindgen(constructor, js_namespace = WebAssembly)]
+        pub fn new(buffer_source: &JsValue) -> Module;
+
+        /// The `WebAssembly.customSections()` function returns a copy of the
+        /// contents of all custom sections in the given module with the given
+        /// string name.
+        ///
+        /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Module/customSections
+        #[wasm_bindgen(static_method_of = Module, js_namespace = WebAssembly, js_name = customSections, catch)]
+        pub fn custom_sections(module: &Module, sectionName: &str) -> Result<Array, JsValue>;
+
+        /// The `WebAssembly.exports()` function returns an array containing
+        /// descriptions of all the declared exports of the given `Module`.
+        ///
+        /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Module/exports
+        #[wasm_bindgen(static_method_of = Module, js_namespace = WebAssembly, catch)]
+        pub fn exports(module: &Module) -> Result<Array, JsValue>;
```

Comment:
Do this, `imports`, and `custom_sections` need `catch`? (or do they regularly return errors?)

Reply:
Right, `catch` can be removed because rust prevents the `TypeError`